internal/impl/devbox.go

```diff
@@ -467,9 +467,18 @@ func (d *Devbox) StartServices(ctx context.Context, serviceNames ...string) erro
 	return nil
 }
 
-func (d *Devbox) StopServices(ctx context.Context, serviceNames ...string) error {
+func (d *Devbox) StopServices(ctx context.Context, allProjects bool, serviceNames ...string) error {
 	if !IsDevboxShellEnabled() {
-		return d.RunScript("devbox", append([]string{"services", "stop"}, serviceNames...))
+		args := []string{"services", "stop"}
+		args = append(args, serviceNames...)
+		if allProjects {
+			args = append(args, "--all-projects")
+		}
```

Comment:
should both `--all-projects` _and_ specific serviceNames be allowed to be specified at the same time?

Should we make this:
```
if allProjects {
   args = append(args, "--all-projects")
   if len(serviceNames) > 0 {
      return errors.New("Cannot specify serviceNames alongwith --all-projects")
   }
} else if len(serviceNames) > 0 {
  args = append(args, serviceNames...)
} else {
   return errors.New("Expected either specific service names or --all-projects")
}
```
?

Reply:
Should we just get the full command and pass it in instead? 

I think we can do:

`d.RunScript("devbox", cmd.CommandPath())`

Reply:
I think `--all-projects` should only work without passing specific service, but we should check that in the boxcli layer.

Reply:
Agreed that if we're going to forbid using --all-projects with service names, we should error on boxcli. I'll implement it that way